Make waitForProjects() wait for ConvertJob

Seen on Travis on #2086
```
Running com.google.cloud.tools.eclipse.appengine.validation.XmlValidatorTest
!ENTRY org.eclipse.wst.jsdt.web.core 4 4 2017-06-26 22:20:37.303
!MESSAGE Resource '/test0.9679344619702539' does not exist.
!STACK 1
org.eclipse.core.internal.resources.ResourceException: Resource '/test0.9679344619702539' does not exist.
	at org.eclipse.core.internal.resources.Resource.checkExists(Resource.java:335)
	at org.eclipse.core.internal.resources.Resource.checkAccessible(Resource.java:209)
	at org.eclipse.core.internal.resources.Project.checkAccessible(Project.java:141)
	at org.eclipse.core.internal.resources.Folder.assertCreateRequirements(Folder.java:33)
	at org.eclipse.core.internal.resources.Folder.create(Folder.java:93)
	at org.eclipse.core.internal.resources.Folder.create(Folder.java:121)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.save(FacetedProject.java:1663)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.setFixedProjectFacetsInternal(FacetedProject.java:545)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.mergeChangesInternal(FacetedProject.java:1209)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.access$2(FacetedProject.java:1117)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject$5.run(FacetedProject.java:1099)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2267)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProject.mergeChanges(FacetedProject.java:1109)
	at org.eclipse.wst.common.project.facet.core.internal.FacetedProjectWorkingCopy.commitChanges(FacetedProjectWorkingCopy.java:2020)
	at org.eclipse.wst.jsdt.web.core.internal.project.ConvertJob.runInWorkspace(ConvertJob.java:66)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:39)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
!SUBENTRY 1 org.eclipse.core.resources 4 368 2017-06-26 22:20:37.322
!MESSAGE Resource '/test0.9679344619702539' does not exist.
Tests run: 7, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 30.463 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.appengine.validation.XmlValidatorTest
testCreateMarker(com.google.cloud.tools.eclipse.appengine.validation.XmlValidatorTest)  Time elapsed: 3.631 sec
testValidate_dynamicWebProject(com.google.cloud.tools.eclipse.appengine.validation.XmlValidatorTest)  Time elapsed: 2.505 sec
testValidate_badXml(com.google.cloud.tools.eclipse.appengine.validation.XmlValidatorTest)  Time elapsed: 0.376 sec  <<< FAILURE!
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.google.cloud.tools.eclipse.appengine.validation.XmlValidatorTest.testValidate_badXml(XmlValidatorTest.java:110)
```

--- a/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlValidatorTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlValidatorTest.java
@@ -23,9 +23,11 @@ import static org.junit.Assert.assertTrue;
 import com.google.cloud.tools.eclipse.appengine.facets.AppEngineStandardFacet;
 import com.google.cloud.tools.eclipse.test.util.project.ProjectUtils;
 import com.google.cloud.tools.eclipse.test.util.project.TestProjectCreator;
+import com.google.common.base.Stopwatch;
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.resources.IMarker;
 import org.eclipse.core.resources.IProject;
@@ -104,7 +106,13 @@ public class XmlValidatorTest {
     IProject project = dynamicWebProjectCreator.getProject();
     IFile file = project.getFile("src/bad.xml");
     file.create(new ByteArrayInputStream(badXml), true, null);
-    ProjectUtils.waitForProjects(project);  // Wait until Eclipse puts an error marker.
+
+    Stopwatch elapsed = Stopwatch.createStarted();
+    while (elapsed.elapsed(TimeUnit.SECONDS) < 300
+        && file.findMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO).length == 0) {
+      ProjectUtils.waitForProjects(project); // Wait until Eclipse puts an error marker.
+    }
+
 
     IMarker[] markers = file.findMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO);
     assertEquals(1, markers.length);

--- a/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlValidatorTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlValidatorTest.java
@@ -108,10 +108,12 @@ public class XmlValidatorTest {
     file.create(new ByteArrayInputStream(badXml), true, null);
 
     Stopwatch elapsed = Stopwatch.createStarted();
-    while (elapsed.elapsed(TimeUnit.SECONDS) < 300
-        && file.findMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO).length == 0) {
+    IMarker[] markers;
+    do {
       ProjectUtils.waitForProjects(project); // Wait until Eclipse puts an error marker.
-    }
+      markers = file.findMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO);
+      System.out.printf("%s: %d problem markers found\n", elapsed, markers.length);
+    } while (elapsed.elapsed(TimeUnit.SECONDS) < 300 && markers.length == 0);
 
 
     IMarker[] markers = file.findMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO);

--- a/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlValidatorTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlValidatorTest.java
@@ -115,8 +115,7 @@ public class XmlValidatorTest {
       System.out.printf("%s: %d problem markers found\n", elapsed, markers.length);
     } while (elapsed.elapsed(TimeUnit.SECONDS) < 300 && markers.length == 0);
 
-
-    IMarker[] markers = file.findMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO);
+    markers = file.findMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO);
     assertEquals(1, markers.length);
 
     String resultMessage = (String) markers[0].getAttribute(IMarker.MESSAGE);

--- a/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlValidatorTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlValidatorTest.java
@@ -21,6 +21,7 @@ import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
 import com.google.cloud.tools.eclipse.appengine.facets.AppEngineStandardFacet;
+import com.google.cloud.tools.eclipse.test.util.ThreadDumpingWatchdog;
 import com.google.cloud.tools.eclipse.test.util.project.ProjectUtils;
 import com.google.cloud.tools.eclipse.test.util.project.TestProjectCreator;
 import com.google.common.base.Stopwatch;
@@ -113,6 +114,9 @@ public class XmlValidatorTest {
       ProjectUtils.waitForProjects(project); // Wait until Eclipse puts an error marker.
       markers = file.findMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO);
       System.out.printf("%s: %d problem markers found\n", elapsed, markers.length);
+      if (markers.length == 0) {
+        ThreadDumpingWatchdog.report("Expected a problem marker", elapsed);
+      }
     } while (elapsed.elapsed(TimeUnit.SECONDS) < 300 && markers.length == 0);
 
     markers = file.findMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO);

--- a/plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/ThreadDumpingWatchdog.java
+++ b/plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/ThreadDumpingWatchdog.java
@@ -137,7 +137,8 @@ public class ThreadDumpingWatchdog extends TimerTask implements TestRule {
     }
     sb.append("\n|");
     dumpEclipseLocks(sb, "| ");
-
+    sb.append("\n|");
+    dumpEclipseJobs(sb, "| ");
     sb.append("\n|");
     int uselessThreadsCount = 0;
     for (ThreadInfo tinfo : infos) {

--- a/plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java
+++ b/plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java
@@ -328,6 +328,7 @@ public class ProjectUtils {
     // some jobs are not part of a family
     for (Job job : jobManager.find(null)) {
       switch (job.getClass().getName()) {
+        case "org.eclipse.wst.jsdt.web.core.internal.project.ConvertJob":
         case "org.eclipse.m2e.core.ui.internal.wizards.ImportMavenProjectsJob":
         case "org.eclipse.m2e.core.internal.project.registry.ProjectRegistryRefreshJob":
           jobs.add(job);

